py-pygobject: add v3.54.3

KineticTheory opened the pull request:
+ Update `url_for_version` function to match other Meson packages.
+ Update dependencies so they apply to the new version.
+ Style fixes.

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

KineticTheory commented:
@rbberger When you have time, can you review this PR? TIA.

rbberger commented on repos/spack_repo/builtin/packages/py_pygobject/package.py:
```diff
@@ -2,44 +2,54 @@
-    depends_on("py-pycairo@1.16:1", type=("build", "run"), when="@3.46.0")
-    depends_on("libffi@3.0.0:", when="@3.46.0")
+    depends_on("python@3.8:3", type=("build", "run"), when="@3.46.0:")
+    depends_on("glib@2.64.0:", when="@3.46.0:")
```
```suggestion
    depends_on("glib@2.64.0:", when="@3.46.0:")
    depends_on("glib@2.80.0:", when="@3.54:")
```

rbberger commented on repos/spack_repo/builtin/packages/py_pygobject/package.py:
```diff
@@ -2,44 +2,54 @@
-    depends_on("libffi@3.0.0:", when="@3.46.0")
+    depends_on("python@3.8:3", type=("build", "run"), when="@3.46.0:")
+    depends_on("glib@2.64.0:", when="@3.46.0:")
+    depends_on("gobject-introspection@1.64.0:", when="@3.46.0:")
```
```suggestion
````
See: https://docs.gtk.org/girepository/migrating-gi.html, it's now part of glib.

KineticTheory committed:
py-pygobject: add v 3.54.3

+ Update url_for_version function to match other Meson packages.
+ Update dependencies so they apply to the new version.
+ When using recent versions of glib, gobject-introspection is no longer needed.
